dockerfiles/ci/centos/6/Dockerfile

```diff
@@ -0,0 +1,81 @@
+FROM centos:6 as base
+
+RUN set -eux; \
+    yum update -y; \
+    yum install -y \
+        centos-release-scl \
+        curl \
+        environment-modules \
+        gcc \
+        gcc-c++ \
+        git \
+        libedit-devel \
+        make \
+        openssl-devel \
+        pkg-config \
+        postgresql-devel \
+        readline-devel \
+        scl-utils \
+        unzip \
+        vim \
+        xz \
+        zlib-devel; \
+    yum install -y devtoolset-7; \
+    yum clean all;
+
+ENV SRC_DIR=/usr/local/src
+
+COPY download-src.sh /root/
+RUN set -eux; \
```

Comment:
This is a good point. As a general rule I personally don't like pulling complete `RUN` commands into a bash script because 1) it's easy to accidentally bust a lot of layer caches at once if the file hash changes for any reason; including changing comments, 2) it's harder to read the Dockerfile when major parts of the build are in a separate file, and 3) it makes things more complicated when things like build args are used.

I see the value of using a separate script when sharing the script across multiple Dockerfiles, but in this case the `base` image acts as the single shared point so we wouldn't be getting that advantage.

That's just my 2 cents. 🙂 Overall I don't have a super strong opinion if we think using a script is better for future builds.